Marlin/Marlin_main.cpp

```diff
@@ -426,6 +426,26 @@ static uint8_t target_extruder;
   int lpq_len = 20;
 #endif
 
+#if ENABLED(HOST_KEEPALIVE_FEATURE)
+
+  // States for managing Marlin and host communication
+  // Marlin sends messages if blocked or busy
+  enum MarlinBusyState {
```

Comment:
This is a brilliant idea: states.
If the "state" list is extended like `HEATING`, `COOLING`, `HOMING`, etc we could now provide good valuable feedback on the LCD screen for the user.. as of today that little status line is pretty useless (mine is currently saying "heating" and my print has already finished). We could also taking this synergy to when a `BUSY/WAITING` reply is sent back to the host, we could send also send back the current sate of Marlin as a flag. Software people could then use this info to produce more accurate _printer status_ on their applications.